### PR TITLE
CLN: remove tseries.util.pivot_annual/isleapyear

### PR DESCRIPTION
**Pull request by topper-123**
``pivot_annual`` and ``isleapyear`` in ``tseries.util`` were deprecated in v0.19.

This PR removes them from the code base.

### CODE REVIEW

File: pandas/tests/reshape/test_pivot.py [hunk 1]
```diff
@@ -1525,116 +1534,3 @@ def test_crosstab_dup_index_names(self):
...
-
-    def test_hourly(self):
-        rng_hourly = date_range('1/1/1994', periods=(18 * 8760 + 4 * 24),
-                                freq='H')
-        data_hourly = np.random.randint(100, 350, rng_hourly.size)
-        ts_hourly = Series(data_hourly, index=rng_hourly)
-
-        grouped = ts_hourly.groupby(ts_hourly.index.year)
-        hoy = grouped.apply(lambda x: x.reset_index(drop=True))
-        hoy = hoy.index.droplevel(0).values
```
Comment (jreback) [inline pandas/tests/reshape/test_pivot.py]: anything worthy of a test for these? eg maybe see if something similar using pivot_table (if not can u construct a test)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/reshape/test_pivot.py (hunk 1 above)
Comment (topper-123) [reply pandas/tests/reshape/test_pivot.py]: A point is that ``pivot_table`` requires dataframes, and ``pivot_annual`` works on series [that need to have a DatetimeIndex]. So ``pivot_table`` is strictly speaking not doing the same as ``pivot_annual``.

A deeper second issue is that ``pivot_annual`` dayofyear number 60 is **always** febr. 29, while for other pandas date functionality day 60 switches between febr. 29 and 1.mars , depending on whether we're not having a leap year or not.

I'll take a look into if something can be moved. However, do you not think that such date functionality is checked other places, i..e. we duplicate datetime tests by adding them to ``TestPivotTable``? See for example ``TestPivotTable.test_pivot_timegrouper``, line 716 for some date pivot_table tests.

**Comment by topper-123** [comment]
I've moved ``test_daily`` and ``test_monthly`` to ``testPivotTable`` from ``TestPivotAnnual``. ``test_hourly`` could be moved as well, but I'll do that tomorrow.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @topper-123

**Comment by jreback** [comment]
@topper-123 ok for a followup to move tests if you want.

**Pull request merged**